c++/src/kj/async.h

```diff
@@ -1093,6 +1098,21 @@ class Executor {
   // As with `executeAsync()`, `func` is always destroyed on the requesting thread, after the
   // executor thread has signaled completion. The return value is transferred between threads.
 
+  template <typename T>
+  PromiseCrossThreadFulfillerPair<T> newPromiseAndCrossThreadFulfiller() const;
+  // Like `newPromiseAndCrossThreadFulfiller()`, but the Promise is tied to the event loop
+  // associated with this Executor. This allows the caller to construct Promises associated with
+  // other threads. The caller will have to arrange to move the constructed Promise to its home
+  // thread somehow in order to await or cancel (destroy) it.
+  //
+  // Note that, unless your threads live forever, this function all but _requires_ some cross-thread
```

Comment:
Hmm, this seems pretty hard to get right. Can we make it be the case that if the promise is destroyed without ever making it to the target EventLoop, or the target EventLoop is destroyed without waiting on the promise, then fulfilling the fulfiller just does nothing? That seems like the natural answer, but I think it means that `XThreadPaf` needs to hold a strong reference to `Executor`.

Reply:
Updated with the change. Indeed I ended up just holding a strong reference in XThreadPaf.

Reply:
https://github.com/capnproto/capnproto/compare/fa48cdbbd5fd9594f6d54ba7df6cc19dd39ad54d..c0bc4f269c1eafc0ad6c5c800e75c7b4dc974eec